tests/src/providers/CMakeLists.txt

```diff
@@ -38,7 +38,7 @@ add_qgis_test(testqgswmsprovider.cpp MODULE provider LINKEDLIBRARIES provider_wm
 
 if (POSTGRES_FOUND)
   add_qgis_test(testqgspostgresexpressioncompiler.cpp MODULE provider LINKEDLIBRARIES provider_postgres_a qgis_core)
-  add_qgis_test(testqgspostgresprovider.cpp MODULE provider LINKEDLIBRARIES provider_postgres_a qgis_core)
+  add_qgis_test(testqgspostgresprovider.cpp MODULE provider LINKEDLIBRARIES provider_postgres_a qgis_core LABELS "POSTGRES")
```

Comment:
As commented, please don't do this -- it prevents running the part of these tests which don't require a postgres instance. Rather split the test file into two components so that we've got one which requires a postgres instance to run and one which is independent.

Reply:
Splitting the test for just one new test seems a bit overkill to me ?
If ENABLE_PGTEST is not set the basic tests in that file STILL run, even if the whole test is labeled as POSTGRES, or am I missing something about this ?

Reply:
I mean, I was always confused by the "ALL_BUT_PROVIDERS" test-batch running a test for the "postgresql provider". Isn't that counter-intuitive ?

Reply:
Unless you pass `-L POSTGRES` to the `ctest` invocation you're still going to run all tests in `testqgspostgresprovider.cpp` not requiring a running PostgreSQL backend even if you built with ENABLE_PGTEST unset with my change, or am I missing something ?

That said, the ENABLE_PGTEST documentation states:

> tests/src/CMakeLists.txt:  set (ENABLE_PGTEST FALSE CACHE BOOL "Enable PostgreSQL provider tests")

To me a test called testqgspostgresprovider.cpp should only run when ENABLE_PGTEST is set, or the cmake macro documentation should be reworded to something like "Enable tests requiring superuser access to a running PostgreSQL cluster" ...

Reply:
@nyalldawson I'd merge this in and file a testsuite issue if you really see one.

Note that also testqgspostgresconn.cpp contains compile-time ENABLE_PGTEST checks and DOES  contain code that can and DO run withouth an accessible PostgreSQL cluster. It's just a label I'm adding, which determines whether the test is run when `-L POSTGRES` or `-LE POSTGRES` are passed to `ctest` -- maybe a new label could be introduced, or some runtime control of those tests, but I don't think this should block this memory-fault fixing PR from being merged.

Reply:
> As commented, please don't do this -- it prevents running the part of these tests which don't require a postgres instance. 

I've asked @lnicola to help me test this and he confirms running part of the test not requiring a postgres instance is still working, when ENABLE_PGTEST is off:

> 321: Totals: 17 passed, 0 failed, 0 skipped, 0 blacklisted, 15ms with ENABLE_PGTEST
> 319: Totals: 16 passed, 0 failed, 0 skipped, 0 blacklisted, 1ms without ENABLE_PGTEST

Ability to ONLY run tests not requiring a PostgreSQL instance while building with ENABLE_PGTEST is indeed not supported at the moment, which is also the case for many other tests. Would you consider filing a ticket to provide such feature ?

Reply:
I've added support for `QGIS_PGTEST_DISABLE_CONNECTION` environment variable in the test, so you can STILL skip the tests requiring a backend connection if you want

Reply:
it's QGIS_PGTEST_SKIP_CONNECTION now, via a macro which I've also proposed in isolation in #51910